File: Data Structures & Algorithms/two-integer-sum/submission-4.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        hash = {}
        for index, value in enumerate(nums):
            diff = target - value
            if diff in hash:
                return [hash[diff], index]
            hash[value] = index
                    


        

File: Data Structures & Algorithms/two-integer-sum-ii/submission-4.go
func twoSum(numbers []int, target int) []int {
	start := 0
	end := len(numbers) - 1

	for sum := numbers[end] + numbers[start]; sum != target; {
		if sum > target {
			end -= 1
		} else if sum < target {
			start += 1
		} else {
			break
		}
		sum = numbers[end] + numbers[start]
	}
	return []int{start+1, end+1}
}
